streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -456,15 +456,21 @@ public long restore(final Map<TaskId, Task> tasks) {
                 // TODO: we always try to restore as a batch when some records are accumulated, which may result in
                 //       small batches; this can be optimized in the future, e.g. wait longer for larger batches.
                 final TaskId taskId = changelogs.get(partition).stateManager.taskId();
-                try {
-                    final Task task = tasks.get(taskId);
-                    final ChangelogMetadata changelogMetadata = changelogs.get(partition);
-                    totalRestored += restoreChangelog(task, changelogMetadata);
-                } catch (final TimeoutException timeoutException) {
-                    tasks.get(taskId).maybeInitTaskTimeoutOrThrow(
-                        time.milliseconds(),
-                        timeoutException
-                    );
+                final Task task = tasks.get(taskId);
```

Comment:
Similarly, given the fix below in line 757, the task should never be null.

Reply:
Why did you move it our of the `try-catch` ?

Wanna re-use `task` in the `catch` block that does call `tasks.get(taskId)` again?

Reply:
Nope, I think it's just for me to write on a diff. It's not intentional.